packages/mobile/src/identity/verification.ts

```diff
@@ -522,7 +522,10 @@ async function isAccountUpToDate(
     accountsWrapper.getDataEncryptionKey(address),
   ])
   return (
-    eqAddress(currentWalletAddress, address) && currentDEK && eqAddress(currentDEK.join(), dataKey)
+    // currentDEK is actually a string instead of an array
+    eqAddress(currentWalletAddress, address) &&
+    currentDEK &&
+    eqAddress([currentDEK].join(), dataKey)
```

Comment:
Not sure, but is it possible to fix the type when `getDataEncryptionKey` is called in line 522 without losing parallelization? Would be nice to be consistent with the other change (and to make it easy to search `getDataEncryptionKey` once the type signature in contractkit is fixed)